parquet-hadoop/src/test/java/org/apache/parquet/hadoop/TestEncryption.java

```diff
@@ -0,0 +1,215 @@
+        .withAlgorithm(ParquetCipher.AES_GCM_CTR_V1)
+        .withFooterKeyID("fk")
+        .withAADPrefix(AADPrefix)
+        .withEncryptedColumns(columnPropertiesMap)
+        .build();
+    encryptionPropertiesList[3] = encryptionProperties;
+    decryptionPropertiesList[3] = decryptionProperties; // Same decryption properties
+
+    // #4  Plaintext footer, default algorithm, key metadata, key retriever, AAD
+    columnProperties0 = ColumnEncryptionProperties.builder("binary_field")
+        .withKey(columnKey0)
+        .withKeyID("ck0")
+        .build();
+    columnProperties1 = ColumnEncryptionProperties.builder("int32_field")
+        .withKey(columnKey1)
+        .withKeyID("ck1")
+        .build();
+    columnPropertiesMap = new HashMap<ColumnPath, ColumnEncryptionProperties>();
+    columnPropertiesMap.put(columnProperties0.getPath(), columnProperties0);
+    columnPropertiesMap.put(columnProperties1.getPath(), columnProperties1);
+    encryptionProperties = FileEncryptionProperties.builder(footerKey)
+        .withFooterKeyID("fk")
+        .withPlaintextFooter()
+        .withAADPrefix(AADPrefix)
+        .withEncryptedColumns(columnPropertiesMap)
+        .build();
+    encryptionPropertiesList[4] = encryptionProperties;
+    decryptionPropertiesList[4] = decryptionProperties; // Same decryption properties
+
+
+    MessageType schema = parseMessageType(
+        "message test { "
+            + "required binary binary_field; "
+            + "required int32 int32_field; "
+            + "required int64 int64_field; "
+            + "required boolean boolean_field; "
+            + "required float float_field; "
+            + "required double double_field; "
+            + "required fixed_len_byte_array(3) flba_field; "
+            + "required int96 int96_field; "
+            + "} ");
+    GroupWriteSupport.setSchema(schema, conf);
+    SimpleGroupFactory f = new SimpleGroupFactory(schema);
+
+    for (int encryptionMode = 0; encryptionMode < numberOfEncryptionModes; encryptionMode++) {
+      System.out.println("MODE: "+encryptionMode);
+      
+      Path file = new Path(root, "m_" + encryptionMode + ".parquet.encrypted");
+      ParquetWriter<Group> writer = new ParquetWriter<Group>(
+          file,
+          new GroupWriteSupport(),
+          UNCOMPRESSED, 1024, 1024, 512, true, false, ParquetWriter.DEFAULT_WRITER_VERSION, conf, 
+          encryptionPropertiesList[encryptionMode]);
+      for (int i = 0; i < 1000; i++) {
+        writer.write(
+            f.newGroup()
+            .append("binary_field", "test" + i)
+            .append("int32_field", 32)
+            .append("int64_field", 64l)
+            .append("boolean_field", true)
+            .append("float_field", 1.0f)
+            .append("double_field", 2.0d)
+            .append("flba_field", "foo")
+            .append("int96_field", Binary.fromConstantByteArray(new byte[12])));
+      }
+      writer.close();
```

Comment:
Consider using try-with-resources. That construct would be less error prone (e.g. closing the writer/reader in case of an exception occurs).